### PR TITLE
feat(cc-logs-application-view): enhance filtering

### LINKED ISSUE #1014: cc-logs: support exact match filtering
As of now, when filtering on log messages, the filter is split by space and each token is searched using an 'AND' boolean expression.

We would like to support exact-match filtering. Maybe by using double quotes around the text.

### LINKED ISSUE #1073: cc-logs: add filtering by regex

### PR DESCRIPTION
**Pull request by pdesoyres-cc**
### what this PR do ?

- adds the ability to filter with a regex
- adds the ability to filter with an exact string

Fixes #1073
Fixes #1014

### how to text

Use the sandbox to test the features.

### CODE REVIEW

**Review by hsablonniere — CHANGES_REQUESTED** [review]
This is so nice and elegant, I like the icons. I create a few notes, nothing much. Well done @pdesoyres-cc !!!! :clap:

File: test/regex-parse.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const regExp = parseRegex('/abc/gimsuy');
+
+    expect(regExp.source).to.eql('abc');
+  });
+
+  it('should include the leading slash into the source when trailing slash is not present', () => {
+    const regExp = parseRegex('/abc');
+
+    expect(regExp.source).to.eql('\\/abc');
+  });
```
[the remark this answers is not in the document]
Comment (pdesoyres-cc) [reply test/regex-parse.test.js:21]: I was wondering too so I went for the less strict solution.

File: src/lib/regex-parse.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+/**
+ * @param {string} input
+ * @return {RegExp}
+ */
+export function parseRegex (input) {
+  const match = input.match(/(\/?)(.+)\1([a-z]*)/i);
```
[the remark this answers is not in the document]
Comment (pdesoyres-cc) [reply src/lib/regex-parse.js:7]: nice, I like that

File: src/lib/regex-parse.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  if (match == null) {
+    throw new Error('invalid regular expression format');
+  }
+
+  const flags = Array.from(new Set(match[3]))
+    .filter((flag) => 'gimsuy'.includes(flag))
+    .join('');
```
[the remark this answers is not in the document]
Comment (pdesoyres-cc) [reply src/lib/regex-parse.js:15]: Could be nice, but the `RegExp` constructor fails when passing invalid flags.

**Review by florian-sanders-cc — APPROVED** [review]
I think the feature looks good and it works well from what I've tested.

Gg :sunglasses:

**Review by roberttran-cc — APPROVED** [review]
LGTM! Very neat PR.

A few non-blocking suggestion/question/feedback.

**Review by HeleneAmouzou — APPROVED** [review]
I did the review with @roberttran-cc , apart from the comments he left, LGTM !

File: test/regex-parse.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  it('should throw exception when regex is invalid', () => {
+    expect(() => parseRegex('a(bc')).to.throw(Error);
+  });
+
+  it('should throw exception when regex starts with a slash but does not finish with a slash', () => {
+    expect(() => parseRegex('/abc')).to.throw(Error);
+  });
+
+  it('should be greedy on slash', () => {
```
[the remark this answers is not in the document]
Comment (pdesoyres-cc) [reply test/regex-parse.test.js:43]: I wanted to make sure that a path with multiple slashes is taken into the expression group

File: src/components/cc-logs-application-view/cc-logs-application-view.js [hunk 5]
```diff
@@ -78,9 +80,11 @@ const MENU_ENTRIES = ['live', 'lastHour', 'last4Hours', 'today', 'yesterday', 'l
  * @typedef {import('./cc-logs-application-view.types.js').Validity} Validity
  * @typedef {import('../cc-logs/cc-logs.types.js').Log} Log
  * @typedef {import('../cc-logs/cc-logs.types.js').MetadataRenderer} MetadataRenderer
- * @typedef {import('lit/directives/ref').Ref<CcInputDate>} RefCcInputDate
- * @typedef {import('lit/directives/ref').Ref<CcLogsControl>} RefCcLogsControl
- * @typedef {import('lit/directives/ref').Ref<CcPopover>} RefCcPopover
+ * @typedef {import('../cc-logs/cc-logs.types.js').LogFilter} LogFilter
```
[the remark this answers is not in the document]
Comment (pdesoyres-cc) [reply src/components/cc-logs-application-view/cc-logs-application-view.js:83]: you're right

**Pull request merged**